projects/lights_control/src/lights_can.c

```diff
@@ -0,0 +1,109 @@
+#include "can.h"
+#include "can_hw.h"
+#include "can_msg_defs.h"
+
+#include "can_unpack_impl.h"
+#include "event_queue.h"
+#include "status.h"
+
+#include "lights_can.h"
+#include "lights_events.h"
+#include "lights_gpio.h"
+
+#define SOMEPORT 0
+#define SOMEPIN 0
+
+#define CASE_ACTION(action)            \
+  case ACTION_##action:                \
+    event_raise(EVENT_##action, data); \
+    break;
+
+#define CAN_RX_ADDR \
+  { 0, 11 }
+
+#define CAN_TX_ADDR \
+  { 0, 12 }
+
+// TODO(ELEC-372): remove these and get the device ID from elsewhere
+#define CAN_DEVICE_ID_FRONT 0
+#define CAN_DEVICE_ID_REAR 0
+
+static BoardType s_boardtype;
+
+static CANStorage s_can_storage;
+static CANRxHandler s_rx_handlers[CAN_NUM_RX_HANDLERS];
+
+static StatusCode prv_rx_handler(const CANMessage *msg, void *context, CANAckStatus *ack_reply);
+
+// TODO(ELEC-372):
+// need to figure out:
+//      1.bitrate for can_settings
+//      2.CAN_TX_ADDR
+//      3.CAN_RX_ADDR
+//      4.msg_id
+void lights_can_init(BoardType boardtype, bool loopback) {
+  CANSettings can_settings = { .bitrate = CAN_HW_BITRATE_125KBPS,
+                               .rx_event = EVENT_CAN_RX,
+                               .tx_event = EVENT_CAN_TX,
+                               .fault_event = EVENT_CAN_FAULT,
+                               .tx = CAN_TX_ADDR,
+                               .rx = CAN_RX_ADDR,
+                               .loopback = loopback };
+  volatile CANMessage rx_msg = { 0 };
```

Comment:
You're probably going to want to move `rx_message` off of the stack here, and make it global (or at the very least, `static`).

Right now, when `lights_can_init`'s stack frame is popped, the `rx_msg` is going to be popped as well. Thus, when `context` is accessed in your CAN RX callback (`prv_rx_handler`), you're going to be dereferencing an invalid pointer and you're probably going to segfault.

Reply:
oh yeah I was wondering if `rx_msg` was supposed to be the storage for something internal to CAN or it's only used in setup time. Thanks, I'll change it.

Reply:
More importantly, why are you passing a CAN message as context? That seems a little odd.

Reply:
Oh I guess I was following the examples in `test_can.c`, and didn't pay much attention to the fact that I don't really need it here.